api-put-object-streaming.go

```diff
@@ -437,15 +437,18 @@ func (c Client) putObjectDo(ctx context.Context, bucketName, objectName string,
 	if err != nil {
 		return ObjectInfo{}, err
 	}
+
+	var eTag string
 	if resp != nil {
 		if resp.StatusCode != http.StatusOK {
 			return ObjectInfo{}, httpRespToErrorResponse(resp, bucketName, objectName)
 		}
+		eTag = trimEtag(resp.Header.Get("ETag"))
```

Comment:
I found that there is situation when resp == nil, calling resp.Header raise panic